### PR TITLE
Pull Request

### PR DESCRIPTION
**Pull request by JereDacosta**
This fixes various bugs such as NullPointerException, layout problems, issue #86...

### CODE REVIEW

File: app/src/main/java/science/apolline/utils/DataExport.kt [hunk 1]
```diff
@@ -67,6 +67,7 @@ object DataExport : AnkoLogger {
                 val fw = FileWriter(createFileName("json"))
                 val gson = GsonBuilder().setPrettyPrinting().create()
                 val jsonFile = gson.toJson(dataList)
+                print(jsonFile.toString())
```
Comment (rouvoy) [inline app/src/main/java/science/apolline/utils/DataExport.kt]: Is this instruction really necessary or a past debugging instruction you introduced?

Comment (JereDacosta) [reply app/src/main/java/science/apolline/utils/DataExport.kt]: It is not necessary, it is indeed a debugging instrcution that I forgot to take away

File: app/src/main/java/science/apolline/view/activity/MainActivity.kt [hunk 2]
```diff
@@ -98,18 +91,18 @@ class MainActivity : RootActivity(), NavigationView.OnNavigationItemSelectedList
...
+                    builder.setPositiveButton("reconnecter") { dialog, which ->
                         registerReceiver(this, MainActivity.makeGattUpdateIntentFilter())
                         if (MainActivity.mBluetoothLeService != null) {
-                            val result = MainActivity.mBluetoothLeService!!.connect(mPrefs.getString("sensor_mac_address","address not found"))
+                            val result = MainActivity.mBluetoothLeService!!.connect(mPrefs.getString("sensor_mac_address", "address not found"))
                             Log.d(MainActivity.TAG, "Connect request result=$result")
                         }
                     }
-                    builder.setNegativeButton("Ok"){dialog,which ->
+                    builder.setNegativeButton("Ok") { dialog, which ->
```
Comment (rouvoy) [inline app/src/main/java/science/apolline/view/activity/MainActivity.kt]: Avoid strings in code, use the [Strings.xml](https://github.com/Apolline-Lille/apolline-android/blob/master/app/src/main/res/values/strings.xml) file instead

File: app/src/main/java/science/apolline/view/activity/MainActivity.kt [hunk 3]
```diff
@@ -227,28 +217,38 @@ class MainActivity : RootActivity(), NavigationView.OnNavigationItemSelectedList
...
-        if (mPrefs.getString("sensor_name" , "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.APPA_REGEX)) {
+        if (mPrefs.getString("sensor_name", "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.APPA_REGEX)) {
             supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#428aff")))
             val gattServiceIntent = Intent(this@MainActivity, BluetoothLeService::class.java)
             bindService(gattServiceIntent, mServiceConnection, Context.BIND_AUTO_CREATE)
         }
         SYNC_MOD = (mPrefs.getString("sync_mod", "2")).toInt()
         INFLUXDB_SYNC_FREQ = (mPrefs.getString("sync_frequency", "60")).toLong()
-    }
+    }*/
```
Comment (rouvoy) [inline app/src/main/java/science/apolline/view/activity/MainActivity.kt]: Why this code is commented ? Should it be removed or not ?

File: app/src/main/java/science/apolline/view/activity/MainActivity.kt [hunk 4]
```diff
@@ -403,15 +405,14 @@ class MainActivity : RootActivity(), NavigationView.OnNavigationItemSelectedList
...
         cancelAutoSync(false)
-        if (mPrefs.getString("sensor_name" , "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.IOIO_REGEX)) {
+        if (mPrefs.getString("sensor_name", "sensor_name does not exist").toLowerCase().contains(regex = SplashScreen.IOIO_REGEX)) {
             stopService(Intent(this, IOIOService::class.java))
-        }
-        else {
+        } else {
             MainActivity.mBluetoothLeService!!.disconnect()
-            unbindService(mServiceConnection)
+            //unbindService(mServiceConnection)
```
Comment (rouvoy) [inline app/src/main/java/science/apolline/view/activity/MainActivity.kt]: Why is that commented?

File: app/src/main/java/science/apolline/view/fragment/ChartFragment.kt [hunk 5]
```diff
@@ -69,6 +69,9 @@ class ChartFragment : RootFragment(), OnChartValueSelectedListener, FragmentLife
                 .build()
     }
 
+    init {
+        info {  "*****TESTING " +this.toString() }
```
Comment (rouvoy) [inline app/src/main/java/science/apolline/view/fragment/ChartFragment.kt]: Should this be kept?

File: app/src/main/java/science/apolline/utils/DataExport.kt (hunk 1 above)
Comment (rouvoy) [reply app/src/main/java/science/apolline/utils/DataExport.kt]: Please remove it and deliver a code as clean as possible

Comment (JereDacosta) [reply app/src/main/java/science/apolline/utils/DataExport.kt]: ok thanks

File: app/src/main/java/science/apolline/view/activity/MainActivity.kt (hunk 3 above)
Comment (JereDacosta) [reply app/src/main/java/science/apolline/view/activity/MainActivity.kt]: Not sure, this was already there before, I can remove it

**Pull request merged**